platform-sdk/swirlds-base/src/main/java/com/swirlds/base/time/Time.java

```diff
@@ -17,46 +17,16 @@
 package com.swirlds.base.time;
 
 import com.swirlds.base.time.internal.OSTime;
-import edu.umd.cs.findbugs.annotations.NonNull;
-import java.time.Instant;
+import java.time.InstantSource;
 
 /**
- * An API for getting the time. All platform code should utilize this API instead of the raw standard
- * java time APIs. This makes it much easier to simulate time in test environments.
+ * @deprecated users of this class should migrate to {@link TimeSource} for measuring elapsed time or {@link InstantSource} for wall-clock.
  */
-public interface Time {
+@Deprecated
```

Comment:
I am not sure we want to deprecate this interface. Keeping `Time` as a single source or all time information is useful, at least in the platform.

Reply:
I agree. We should only deprecate this class if we intend to remove it. In the past we have been over eager to deprecate classes without the corresponding drive to actually remove them. The result of this is that we end up with a lot of deprecated, which is ugly.

Reply:
That is very true; my aim here was to start the discussion. It depends on our goals... Having time to implement the two interfaces makes solving the drawbacks mentioned in the ticket easier, but we wouldn't be replacing anything, and it invites one to think if the whole change makes sense. The other option is to remove time, and to modify the mentioned classes that use both TimeSource and InstantSource to receive both parameters/fields, but that also seems ugly and with the latest change @cody-littley  introduced, it became a broader change.